test/celeritas/optical/VolumeSurfaceSelector.test.cc

```diff
@@ -0,0 +1,225 @@
+
+        VolumeSurfaceSelector select{surfaces_->host_ref(),
+                                     volumes_->volume(pre_vol_inst),
+                                     pre_vol_inst};
+        for (auto post_vol_inst :
+             range(VolumeInstanceId{volumes_->num_volume_instances()}))
+        {
+            if (post_vol_inst != pre_vol_inst)
+            {
+                results.push_back(
+                    select(volumes_->volume(post_vol_inst), post_vol_inst));
+            }
+        }
+
+        return results;
+    }
+
+    std::shared_ptr<VolumeParams const> volumes_;
+    std::shared_ptr<SurfaceParams const> surfaces_;
+};
+
+//---------------------------------------------------------------------------//
+// TESTS
+//---------------------------------------------------------------------------//
+// Test surface selection for various pre and post volume instances
+TEST_F(VolumeSurfaceSelectorTest, select_surface)
+{
+    if (CELERITAS_CORE_GEO == CELERITAS_CORE_GEO_GEANT4)
+    {
+        {
+            static SurfaceId const expected_surfaces[] = {
+                SurfaceId{0},
+                SurfaceId{0},
+                SurfaceId{0},
+                SurfaceId{0},
+            };
+
+            EXPECT_VEC_EQ(expected_surfaces,
+                          this->select_surfaces(VolumeInstanceId{0}));
+        }
+        {
+            static SurfaceId const expected_surfaces[] = {
+                SurfaceId{1},
+                SurfaceId{2},
+                SurfaceId{1},
+                SurfaceId{1},
+            };
+
+            EXPECT_VEC_EQ(expected_surfaces,
+                          this->select_surfaces(VolumeInstanceId{1}));
+        }
+        {
+            static SurfaceId const expected_surfaces[] = {
+                SurfaceId{0},
+                SurfaceId{3},
+                SurfaceId{4},
+                SurfaceId{},
+            };
+
+            EXPECT_VEC_EQ(expected_surfaces,
+                          this->select_surfaces(VolumeInstanceId{2}));
+        }
+        {
+            static SurfaceId const expected_surfaces[] = {
+                SurfaceId{1},
+                SurfaceId{1},
+                SurfaceId{1},
+                SurfaceId{1},
+            };
+
+            EXPECT_VEC_EQ(expected_surfaces,
+                          this->select_surfaces(VolumeInstanceId{3}));
+        }
+        {
+            static SurfaceId const expected_surfaces[] = {
+                SurfaceId{0},
+                SurfaceId{1},
+                SurfaceId{},
+                SurfaceId{1},
+            };
+
+            EXPECT_VEC_EQ(expected_surfaces,
+                          this->select_surfaces(VolumeInstanceId{4}));
+        }
+    }
+}
+
+//---------------------------------------------------------------------------//
+// Check selector correctly uses volume instances from a geo view
+TEST_F(VolumeSurfaceSelectorTest, geo_view_wrapper)
+{
+    if (CELERITAS_CORE_GEO == CELERITAS_CORE_GEO_GEANT4)
```

Comment:
Even if we did need the g4 geo, you should disable this at the cmake level: otherwise it looks like this test is running and passing when it's not doing anything at all.